pkg/pillar/hypervisor/kvm.go

```diff
@@ -456,6 +462,8 @@ func (ctx kvmContext) Setup(status types.DomainStatus, config types.DomainConfig
 	diskStatusList := status.DiskStatusList
 	domainName := status.DomainName
 	domainUUID := status.UUIDandVersion.UUID
+	// Set the CPUs in the local copy of domain DomainConfig, as it's used to create the config file
+	config.VmConfig.CPUs = status.VmConfig.CPUs
```

Comment:
This doesn't necessarily work, since config will get update if there is a handleModify due to some unrelated change published by zedmanager.

Can not the config be generated using DomainConfig plus DomainStatus? 
If not you have to make sure that handleModify alwayts copies back status.VmComnfig.CPUs to config, which is a very odd pattern.

Reply:
Moving CPUs from VmConfig to DomainStatus might make this more obvious, unless we think that the list of CPUs will be part of what we receive from the controller in the future. But here we determine the list of CPUs hence having it in DomainStatus and not VmConfig seems to fit better.

Reply:
Currenly CreateDomConfig() takes a DiskStatusList to handle the case of update disks from the config (e.g., the mounts, the cloud-init iso). Making CreateDomConfig either take a DomainStatus and have it get DiskStatusList from there, and get CPUs from DomainStatus, OR explicitly passing the CPUs argument to CeateDomConfig() would make this easier.

Reply:
Ok, I'll check it. Previously I didn't find an easy way to add the corresponding line into the template with Status. Looks like the template handling happens only with the Config. I'll take a closer look now.

Reply:
Ok, I added the DomainStatus parameter to the `CreateDomConfig()` function. Works fine. There are no writes to the DomainConfig now =)